tests/feature/test_steps.py

```diff
@@ -1,84 +1,238 @@
+                    Given I have a foo fixture with value "foo"
+                    Then foo should have value "foo"
+
+            """
+        ),
+    )
+    testdir.makepyfile(
+        textwrap.dedent(
+            """\
+        from pytest_bdd import given, then, scenario
 
+        @scenario("steps.feature", "Then step can follow Given step")
+        def test_steps():
+            pass
 
-@scenario("steps.feature", "All steps are declared in the conftest")
-def test_conftest():
-    pass
+        @given('I have a foo fixture with value "foo"')
+        def foo():
+            return "foo"
 
+        @then('foo should have value "foo"')
+        def foo_is_foo(foo):
+            assert foo == "foo"
 
-def test_multiple_given(request):
+        """
+        )
+    )
+    result = testdir.runpytest()
+    assert result.ret == 0
+
+
+def test_conftest(testdir):
+    testdir.makefile(
+        ".feature",
+        steps=textwrap.dedent(
+            """\
+            Feature: Steps are executed one by one
+                Steps are executed one by one. Given and When sections
+                are not mandatory in some cases.
+
+                Scenario: All steps are declared in the conftest
+                    Given I have a bar
+                    Then bar should have value "bar"
+
+            """
+        ),
+    )
+    testdir.makeconftest(textwrap.dedent(
+        """\
+        from pytest_bdd import given, then
+
+
+        @given("I have a bar")
+        def bar():
+            return "bar"
+
+
+        @then('bar should have value "bar"')
+        def bar_is_bar(bar):
+            assert bar == "bar"
+
+        """
+    ))
+    testdir.makepyfile(
+        textwrap.dedent(
+            """\
+        from pytest_bdd import scenario
+
+        @scenario("steps.feature", "All steps are declared in the conftest")
+        def test_steps():
+            pass
+
+        """
+        )
+    )
+    result = testdir.runpytest()
+    assert result.ret == 0
+
+
+def test_multiple_given(request, testdir):
     """Using the same given fixture raises an error."""
+    feature = testdir.makefile(
+        ".feature",
+        steps=textwrap.dedent(
+            """\
+            Feature: Steps are executed one by one
+                Steps are executed one by one. Given and When sections
+                are not mandatory in some cases.
+
+                Scenario: Using the same given fixture raises an error
+                    Given I have a bar
+                    And I have a bar
+
+            """
+        ),
+    )
+    testdir.makeconftest(textwrap.dedent(
+        """\
+        from pytest_bdd import given
+
+        @given("I have a bar")
+        def bar():
+            return "bar"
+
+        """
+    ))
 
-    @scenario("steps.feature", "Using the same given fixture raises an error")
+    @scenario(feature.strpath, "Using the same given fixture raises an error")
     def test():
```

Comment:
also here we should use the tesdtir.runpytest way. the `RunResult` class allows to match lines:
`result.stderr.fnmatch_lines(["*given step is already used*"])`